src/components/EditTextModal/EditTextModal.js

```diff
@@ -0,0 +1,158 @@
+  ]);
+
+  const [t] = useTranslation();
+  const dispatch = useDispatch();
+  const [newText, setNewText] = useState('');
+  // Note: core.getSelectedTextQuads() to store selected text coordinates in the following format
+  // ie. {1:[{x1: 81, x2: 553, x3: 553, x4: 81, y1: 608, y2: 608, y3: 592, y4: 592, }, {...}]}
+  // when storing in textCoordinates, we take off the wrapper object and only store array
+  const [textCoordinates, setTextCoordinates] = useState([]);
+
+  const closeModal = () => {
+    dispatch(actions.closeElement('editTextModal'));
+    setNewText('');
+    core.setToolMode(defaultTool);
+  };
+
+  useEffect(
+    () => {
+
+      const storeConvertedPDFCoordinates = () => {
+        // core.getSelectedTextQuads() returns 4 coordinates pair
+        // (x4,y4)---(x3,y3)
+        //   |          |
+        // (x1,y1)---(x2,y2)
+        // Firstly convert it to PDF coordinates and then store it in a flat structure
+        const textCoordinates = core.getSelectedTextQuads();
+        // Example:
+        // textCoordinates = {
+        //   1: [
+        //     {x1: 32, y1: 582, x2: 67.28, y2: 582, x3: 67.28,4: 32, y1: 582, y2: 582, y3: 565, y4: 565},
+        //     {x1: 32, x2: 68, x3: 68, x4: 32, y1: 595, y2: 595, y3: 579, y4: 579},
+        //   ]
+        // }
+        // convert to a flat structure
+        // textCoordinates = [{x1:32, x2:1, y1:582, y2:2, pageNum:1}, {x1:1, x2:1, y1:2, y2:2, pageNum:1}]
+
+        if (textCoordinates) {
+          const convertedTextCoordinates = [];
+          Object.keys(textCoordinates).forEach(pageNum => {
+            textCoordinates[pageNum].forEach(rect => {
+              const point1 = window.docViewer.getDocument().getPDFCoordinates(pageNum, rect.x1, rect.y1);
+              const point2 = window.docViewer.getDocument().getPDFCoordinates(pageNum, rect.x3, rect.y3);
+              convertedTextCoordinates.push({ x1: point1.x, x2: point2.x, y1: point1.y, y2: point2.y,pageNum:Number(pageNum) });
+            });
+          });
+          setTextCoordinates(convertedTextCoordinates);
+        }
+      };
+
+      if (isOpen) {
+        //  prepopulate selected text
+        setNewText(core.getSelectedText());
+        // convert and store selected text coordinates
+        storeConvertedPDFCoordinates();
+      }
+    },
+    [isOpen]
+  );
+
+
+
+  const updateText = e => {
+    e.preventDefault();
+    window.PDFNet.runWithCleanup(async() => {
+      const doc = await window.docViewer.getDocument().getPDFDoc();
+      // Treat each line as a rectangle region and update them separately
+      const lines = newText.split(/\r?\n/);
+      doc.initSecurityHandler();
+      doc.lock();
+      const PDFNet = window.PDFNet;
+      // Note: If the number of lines are more than number of rects defined in textCoordinates, then we just ignore the rest
+      if (textCoordinates && textCoordinates.length > 0) {
+        for (let i=0; i<textCoordinates.length; i++) {
```

Comment:
Character spacing should be like this? `let i = 0; i < textCoordinates.length; i++)`

This isn't in our linting currently?

Reply:
It might be in our linting, but unless devs have eslint integrated in their IDEs they might miss it

Reply:
I have eslint integrated with my IDE this section of code has been formatted.